### PR TITLE
SPEC-975, SPEC-977: Add test cases for: UTF-8 char, unknown key, wrong value type

### PR DESCRIPTION
**Pull request by derickr**

### CODE REVIEW

**Comment by derickr** [comment]
@ShaneHarvey Done - I've also added a test for two host names in the URI too, and clarified the text saying so.

File: source/initial-dns-seedlist-discovery/tests/README.rst [hunk 1]
```diff
@@ -23,11 +23,17 @@ these tests::
...
   _mongodb._tcp.test5.test.build.10gen.cc.  86400  IN SRV  27017  localhost.build.10gen.cc.
   _mongodb._tcp.test6.test.build.10gen.cc.  86400  IN SRV  27017  localhost.build.10gen.cc.
+  _mongodb._tcp.test7.test.build.10gen.cc.  86400  IN SRV  27017  localhost.build.10gen.cc.
+  _mongodb._tcp.test8.test.build.10gen.cc.  86400  IN SRV  27017  localhost.build.10gen.cc.
 
   Record                                    TTL    Class   Text
   test5.test.build.10gen.cc.                86400  IN TXT  "connectTimeoutMS=300000&socketTimeoutMS=300000"
   test6.test.build.10gen.cc.                86400  IN TXT  "connectTimeoutMS=200000"
   test6.test.build.10gen.cc.                86400  IN TXT  "socketTimeoutMS=200000"
+  test7.test.build.10gen.cc.                86400  IN TXT  "readPreference=secondaryPreferred&readPreferenceTags=🥃"
```
Comment (ajdavis) [inline source/initial-dns-seedlist-discovery/tests/README.rst]: Brilliant.

Comment (derickr) [reply source/initial-dns-seedlist-discovery/tests/README.rst]: Picked it over a normal smiley, as a smiley is part of the BMP, but this glass is not.

**Comment by ShaneHarvey** [comment]
@behackett are we able to test this before it is merged?

**Comment by derickr** [comment]
@ShaneHarvey I don't think so, I need to create a BUILD/OFFICE-IT ticket for them to add the DNS records once everybody approved this (and PR #234).

**Comment by ShaneHarvey** [comment]
Okay thanks, still LGTM.

**Comment by derickr** [comment]
@mbroadst and @behackett, still waiting on a LGTM from you.

**Pull request merged**